src/content_script/input_tools.js

```diff
@@ -0,0 +1,237 @@
+      if (!InputTools.isValidEmailInput(element)) {
+        return;
+      }
+
+      // ignore if this elements has already been tracked
+      const i = InputTools.trackedElements.indexOf(element);
+      if (i === -1) {
+        InputTools.trackedElements.push(element);
+        InputTools.addSLButtonToInput(element);
+      }
+    });
+  },
+
+  destroy(target) {
+    InputTools.queryEmailInputAndApply(target, (element) => {
+      // remove element from tracking list
+      const i = InputTools.trackedElements.indexOf(element);
+      if (i !== -1) {
+        InputTools.trackedElements.splice(i, 1);
+      }
+    });
+  },
+
+  queryEmailInputAndApply(target, actionFunction) {
+    if (!target.querySelectorAll) return;
+    const elements = target.querySelectorAll(
+      "input[type='email'],input[name*='email'],input[id*='email']"
+    );
+    for (const element of elements) {
+      actionFunction(element);
+    }
+  },
+
+  isValidEmailInput(element) {
+    const style = getComputedStyle(element);
+    return (
+      // check if element is visible
+      style.visibility !== "hidden" &&
+      style.display !== "none" &&
+      style.opacity !== "0" &&
+      // check if element is not disabled
+      !element.disabled &&
+      // for example, we must filter out a checkbox with name*=email
+      // check if element is text or email input
+      (element.type === "text" || element.type === "email")
+    );
+  },
+
+  addSLButtonToInput(inputElem, isInputFixed) {
+    // create wrapper for SL button
+    const btnWrapper = InputTools.newDiv("sl-button-wrapper");
+    const inputSumHeight = inputElem.getBoundingClientRect().height + "px";
+    btnWrapper.style.height = inputSumHeight;
+    btnWrapper.style.width = inputSumHeight;
+    document.body.appendChild(btnWrapper);
+
+    // create the SL button
+    const slButton = InputTools.newDiv("sl-button");
+    slButton.addEventListener("click", function () {
+      InputTools.handleOnClickSLButton(inputElem, slButton);
+    });
+    slButton.style.height = inputSumHeight;
+    slButton.style.width = inputSumHeight;
+    btnWrapper.appendChild(slButton);
+
+    InputTools.placeElementToTheRight(inputElem, btnWrapper, isInputFixed);
+  },
+
+  newDiv(classes) {
+    const div = document.createElement("div");
+    div.classList.add(classes);
+    return div;
+  },
+
+  placeElementToTheRight(anchor, elem, isFixed) {
+    let intervalId = 0;
+
+    function updatePosition() {
+      // check is element is removed from trackedElements
+      const i = InputTools.trackedElements.indexOf(anchor);
+      if (i === -1) {
+        elem.parentNode.removeChild(elem);
+        clearInterval(intervalId);
+      }
+
+      // get dimension & position of anchor
+      const anchorCoords = anchor.getBoundingClientRect();
+      const anchorStyle = getComputedStyle(anchor);
+      const elemWidth = InputTools.dimensionToInt(elem.style.width);
```

Comment:
should we pass the width of `elem` here which is `inputSumHeight` in `addSLButtonToInput`? I wonder if we can remove `dimensionToInt` actually

Reply:
If we decide not to re-use `placeElementToTheRight` in other place, then yes we can use `inputSumHeight` here

Reply:
fixed on cd4dd7e19e3a95cc8753ec48f2364eab409e66da